packages/nextjs-mf/src/plugins/container/runtimePlugin.ts

```diff
@@ -43,11 +61,29 @@ export default function (): FederationRuntimePlugin {
       // console.log('loadRemote: ', args);
       return args;
     },
-    async loadShare(args) {
+    loadShare(args) {
       // console.log('loadShare:', args);
+      return args;
     },
     async beforeLoadShare(args) {
-      // console.log('beforeloadShare:', args);
+      if (!args.pkgName === 'react' || !args.pkgName === 'react-dom')
+        return args;
+
+      args.shareInfo.strategy = function ({
+        shareScopeMap,
+        scope,
+        pkgName,
+        findVersion,
+      }) {
+        const host = globalThisVal['__FEDERATION__']['__INSTANCES__'][0]; // get root host instance
+        const found = findVersion(shareScopeMap, scope, pkgName);
+        // if return function, then it skips findVersion key, if return key, it will use that to look up in shareScopeMap
+        return function ({ localShareScopeMap, sc, pkgName }) {
```

Comment:
This needs to be implemented, better, we ideally should have 1 or 2 other hooks as this is bit of a elaborate workaround to springbosrd into the api with existing hook